pkg/kubelet/node_labels.go

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/golang/glog"
+	"io/ioutil"
+	"k8s.io/kubernetes/pkg/util"
+	"k8s.io/kubernetes/pkg/util/fielderrors"
+	"path"
+	"path/filepath"
+	"regexp"
+	"sync"
+	"time"
+)
+
+const pollPluginDuration = time.Minute * 10
+const validNameSpaces = `^.+\.node.kubernetes.io/.+$`
+
+var validNameSpacesRegex = regexp.MustCompile(validNameSpaces)
+
+type NodeLabelManager interface {
+	Run()
+	GetNodeLabels() map[string]string
+}
+
+type nodeLabelManager struct {
+	labels    map[string]string
+	hostname  string
```

Comment:
instead of `hostname` would this be better represented as `defaultLabels map[string]string` that the caller (kubelet) provides?